lib/ansible/modules/cloud/docker/docker_node.py

```diff
@@ -0,0 +1,290 @@
+       (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+       For Python 2.6, C(docker-py) must be used. Otherwise, it is recommended to
+       install the C(docker) Python module. Note that both modules should I(not)
+       be installed at the same time. Also note that when both modules are installed
+       and one of them is uninstalled, the other might no longer function and a
+       reinstall of it is required."
+    - "The docker server >= 1.10.0"
+author:
+  - Piotr Wojciechowski (@wojciechowskipiotr)
+  - Thierry Bouvet (@tbouvet)
+
+'''
+
+EXAMPLES = '''
+- name: List names of registered swarm nodes
+  docker_node:
+    state: list
+
+- name: Set node role
+  docker_node:
+    state: update
+    name: mynode
+    role: manager
+
+- name: Set node availability
+  docker_node:
+    state: update
+    name: mynode
+    availability: drain
+
+- name: Replace node labels with new labels
+  docker_node:
+    state: update
+    name: mynode
+    labels:
+      key: value
+    labels_state: replace
+
+- name: Merge node labels and new labels
+  docker_node:
+    state: update
+    name: mynode
+    labels:
+      key: value
+
+- name: Merge node labels and new labels
+  docker_node:
+    state: update
+    name: mynode
+    labels:
+      key: value
+    labels_state: merge
+
+- name: Remove node labels
+  docker_node:
+    state: update
+    name: mynode
+    labels_state: remove
+'''
+
+RETURN = '''
+node_facts:
+  description: Information about node after 'update' operation
+  returned: success
+  type: dict
+actions:
+  description: Provides the actions done on the swarm.
+  returned: when action failed.
+  type: list
+  example: "['This cluster is already a swarm cluster']"
+
+'''
+
+try:
+    from docker.errors import APIError
+except ImportError:
+    # missing docker-py handled in ansible.module_utils.docker_common
+    pass
+
+from ansible.module_utils.docker_common import (
+    DockerBaseClass,
+)
+
+from ansible.module_utils._text import to_native
+
+from ansible.module_utils.docker_swarm import AnsibleDockerSwarmClient
+
+
+class TaskParameters(DockerBaseClass):
+    def __init__(self, client):
+        super(TaskParameters, self).__init__()
+
+        self.state = None
+
+        # Spec
+        self.name = None
+        self.labels = None
+        self.labels_state = None
+
+        # Node
+        self.availability = None
+        self.role = None
+
+        for key, value in client.module.params.items():
+            setattr(self, key, value)
+
+
+class SwarmNodeManager(DockerBaseClass):
+
+    def __init__(self, client, results):
+
+        super(SwarmNodeManager, self).__init__()
+
+        self.client = client
+        self.results = results
+        self.check_mode = self.client.check_mode
+
+        self.parameters = TaskParameters(client)
+
+    def __call__(self):
+        choice_map = {
+            "update": self.node_update,
+        }
+
+        choice_map.get(self.parameters.state)()
```

Comment:
Why not `choice_map[self.parameters.state]()`? That's shorter, and if it doesn't find the key, it will crash anyway.

Reply:
Except for being shorter is there any other advantage? In `docker_swarm` we have `.get()` as well, in other modules we have simple conditionals.

Reply:
Why use a longer / more complicated form when the shorter also works? :)

Reply:
> Why use a longer / more complicated form when the shorter also works, and improves readability? :)

Reply:
Tbh, in this case, in my opinion, it is not an improvement just preferred way of coding. but I don't mind changing it :)